Source/Common/Config.cpp

```diff
@@ -53,11 +53,18 @@ namespace FEX::Config {
       Program += ProgramFDFromEnv;
     }
 
+    // If we were provided a relative path then we need to canonicalize it to become absolute.
+    // If the program name isn't resolved to an absolute path then glibc breaks inside it's `_dl_get_origin` function.
+    // This is because we rewrite `/proc/self/exe` to the absolute program path calculated in here.
+    if (!Program.starts_with('/')) {
+      Program = std::filesystem::canonical(Program).string();
+    }
+
     // If FEX was invoked through an FD path (either binfmt_misc or execveat) then we need to check the
     // Program to see if it is a symlink to find the real path.
     //
-    // binfmt_misc: Arg[0] is actually the full application path or `/dev/fd/<FD>` path
-    //   - Full application path with execve (See Side Note)
+    // binfmt_misc: Arg[0] is actually the execve `pathname` or `/dev/fd/<FD>` path
```

Comment:
It's not really clear what `pathname` refers to.

Reply:
Would `execve pathname argument` work?

Reply:
Ah yes, that would be fine.

Reply:
Switched it to that